segments/battery.p9k

```diff
@@ -77,38 +77,47 @@ prompt_battery() {
   if [[ "$__P9K_OS" == 'Linux' ]] || [[ "$__P9K_OS" == 'Android' ]]; then
     # needed if files or directory do not match globbing pattern
     local sysp="${ROOT_PREFIX}/sys/class/power_supply"
-    local -a bats
+    declare -a bats
     bats=( $sysp/(BAT*|battery)(N) )
     [[ ${#bats} == 0 ]] && return
 
     local energy_now="0"
     local energy_full="0"
     local power_now="0"
     local battery_status_full=true
+    local battery_status_calculating=false
     local battery_status_charging=false
     for bat in $bats; do
       # skip this loop if no files in $bat
       local -a bat_tmp
       bat_tmp=( $bat/*(N) )
       [[ ${#bat_tmp} == 0 ]] && continue
       # add "+ 0[number in file]" if the file exists (0 padding if glob fails)
-      local -a energy_now_tmp
+      declare -a energy_now_tmp
```

Comment:
What is the advantage to use `declare -a` vs `local -a`?

Reply:
Didn't you tell me about it being compatible with zsh version 5.0 and it would be nice if most of the code would be compatible even if not officially supported? Did I misunderstand that?
I'll happily get rid of the extra lines.

Also, apparently I missed L92.

Reply:
Yes I did. But is `local -a` invalid in ZSH 5.0? AFAIR only declaring and setting a value in one line is illegal.